library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -129,12 +281,22 @@ public void setConnectionFactoryClass(String connectionFactoryClass)
    */
   public Map<String, String> getConnectionFactoryProperties()
   {
-    return connectionFactoryProperties;
+    DefaultConnectionFactoryBuilder builder =
```

Comment:
This method should also behave like getConnectionFactoryClass, throw Unsupported exception and not create a default factory builder as it is a getter. The convenience methods below setUser etc may not be able to use this method then but their functionality is correct.